### ISSUE TITLE
Propegate the build order up on a project if a project is PackageReferenced but set on each one to not reference it's outputs

### ISSUE THREAD

**Issue by AraHaan** [dotnet/msbuild#6123]
### Issue Description

Based on my information I sent on my previous issue #6122, I noted there is an issue where my build tasks get built not first, but almost last in the build tree when invoking the build command for the 1st time when the Task Assemblies do not exist yet. In this case it would make build fail on all of the other projects from before it because of the fact it's not set to reference the assembly of the build task on those projects because I was planning on using the ProjectReferences to boost the build order of the task up to 1st since all of the projects depend on it in my solution.

### Steps to Reproduce

1. Make an simple msbuild task project with c# code that could do anything including printing stuff to msbuild's output handle.
2. add other projects to the solution file that then ProjectReferences that build task imports an props file from the task project on how to find the task assemblies based on the build output directories of the actual task assemblies.
3. build and the task(s) would be built last resulting in every single one of the other projects to fail.

### Expected Behavior

For the task assemblies to be built on the TFM's to target.

### Actual Behavior

The task assemblies get built last or next to last.

### Analysis

Not sure where the issue is.

### Versions & Configurations

Windows 10 latest insider dev channel build (x64).
.NET SDK 5.0.102 and that version of msbuild.

### Attach a binlog

This should be easily reproducible by anyone that basically anyone can make a binlog for it.

**Comment by AraHaan** [comment]
It should be reproducible with this commit on the the repository I experienced it normally before too.

https://github.com/Elskom/Sdk/tree/aa74e9e2037c834c463d321cd1983fbdc4ddfd26

I have since moved it to a metapackage, and changed the build task to an inline ``*.targets`` file that then gets invoked to work around this issue and #6122 (however I still get issues sometimes with it failing on the xml docs a lot of times but it happens a lot less now having the build task inline).

basically if you just build it right from checking out that commit, it should fail by building most of everything first before it even gets to the build task (if it does not build the build task last) when the build task should have been first since every project on that commit references it, just that it does not list it or even use it as a dependency.

**Comment by BenVillalobos** [comment]
Team Triage: What is the task that's failing to build in time?

**Comment by AraHaan** [comment]
It was both Tasks that was in that assembly on that commit in that codebase that I have since migrated off of a project being referenced and instead have it provided with a source generator package instead and automatically ran there when they install the source generator package.

**Comment by Forgind** [comment]
Team triage:
Your task is running before PrepareForBuild. That's also before the ResolveProjectReferences target, which is what causes ProjectReferences to be built. The fix for this is to have your task run after that instead. Alternatively, you can build your repo in graph mode (/graph), since that would cause referenced projects to be built before referencing projects.
